dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1590,23 +1568,31 @@ private File saveInputStreamInTempFile(InputStream inputStream, Long fileSizeLim
             throws IOException, FileExceedsMaxSizeException {
         Path tempFile = Files.createTempFile(Paths.get(getFilesTempDirectory()), "tmp", "upload");
 
-        if (inputStream != null && tempFile != null) {
-            Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
+        FileOutputStream outStream = new FileOutputStream(tempFile.toFile());
+        logger.info("Will attempt to save the file as: " + tempFile.toString());
 
-            // size check:
-            // (note that "no size limit set" = "unlimited")
-            Long fileSize = tempFile.toFile().length();
-            if (fileSizeLimit != null && fileSize > fileSizeLimit) {
+        byte[] buffer = new byte[8 * 1024];
+        int bytesRead;
+        long totalBytesRead = 0;
+        while ((bytesRead = inputStream.read(buffer)) != -1) {
+            totalBytesRead += bytesRead;
+            if (fileSizeLimit != null && totalBytesRead > fileSizeLimit) {
                 try {
                     tempFile.toFile().delete();
```

Comment:
I'm not sure, but it seems to me that we should try to close out stream before deletion.

Reply:
I think closing it in finally clause is sufficient.

Reply:
I thought not about sufficiency but about something different: if file is deleted while we have open stream to that file, then can we safel close that stream (ie. without exception)?
But I think it should be ok.

Reply:
Finally clause is executed even if exception is thrown inside try block